jans-cedarling/cedarling/src/jwt/token.rs

```diff
@@ -0,0 +1,264 @@
+use serde_json::Value;
+use std::collections::HashMap;
+
+const DEFAULT_USER_ID_SRC_CLAIM: &str = "sub";
+const DEFAULT_ROLE_SRC_CLAIM: &str = "role";
+
+pub enum TokenStr<'a> {
+    Access(&'a str),
+    Id(&'a str),
+    Userinfo(&'a str),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Token<'a> {
+    Access(TokenData<'a>),
+    Id(TokenData<'a>),
+    Userinfo(TokenData<'a>),
+}
+
+impl Token<'_> {
+    pub fn kind(&self) -> TokenKind {
+        match self {
+            Token::Access(_) => TokenKind::Access,
+            Token::Id(_) => TokenKind::Id,
+            Token::Userinfo(_) => TokenKind::Userinfo,
+        }
+    }
+
+    pub fn metadata(&self) -> &TokenEntityMetadata {
+        match self {
+            Token::Access(data) => data.iss.unwrap_or_default().tokens_metadata().access_tokens,
+            Token::Id(data) => data.iss.unwrap_or_default().tokens_metadata().access_tokens,
+            Token::Userinfo(data) => data.iss.unwrap_or_default().tokens_metadata().access_tokens,
+        }
+    }
+
+    pub fn user_mapping(&self) -> &str {
+        match self {
+            Token::Access(data) => data
+                .iss
+                .unwrap_or_default()
+                .user_mapping(TokenKind::Access)
+                .unwrap_or(DEFAULT_USER_ID_SRC_CLAIM),
+            Token::Id(data) => data
+                .iss
+                .unwrap_or_default()
+                .user_mapping(TokenKind::Id)
+                .unwrap_or(DEFAULT_USER_ID_SRC_CLAIM),
+            Token::Userinfo(data) => data
+                .iss
+                .unwrap_or_default()
+                .user_mapping(TokenKind::Userinfo)
+                .unwrap_or(DEFAULT_USER_ID_SRC_CLAIM),
+        }
+    }
+
+    pub fn claim_mapping(&self) -> &ClaimMappings {
+        match self {
+            Token::Access(data) => &data.iss.unwrap_or_default().access_tokens.claim_mapping,
+            Token::Id(data) => &data.iss.unwrap_or_default().id_tokens.claim_mapping,
+            Token::Userinfo(data) => &data.iss.unwrap_or_default().userinfo_tokens.claim_mapping,
+        }
+    }
+
+    pub fn role_mapping(&self) -> &str {
+        match self {
+            Token::Access(data) => data
+                .iss
+                .unwrap_or_default()
+                .role_mapping(TokenKind::Access)
+                .unwrap_or(DEFAULT_ROLE_SRC_CLAIM),
+            Token::Id(data) => data
+                .iss
+                .unwrap_or_default()
+                .role_mapping(TokenKind::Id)
+                .unwrap_or(DEFAULT_ROLE_SRC_CLAIM),
+            Token::Userinfo(data) => data
+                .iss
+                .unwrap_or_default()
+                .role_mapping(TokenKind::Userinfo)
+                .unwrap_or(DEFAULT_ROLE_SRC_CLAIM),
+        }
+    }
+
+    pub fn get_claim(&self, name: &str) -> Option<TokenClaim> {
+        match self {
+            Token::Access(data) | Token::Id(data) | Token::Userinfo(data) => data.get_claim(name),
```

Comment:
This check has no sense

Reply:
and others like this

Reply:
in each case where `.get_claim()` is used we can use `&TokenData`

Reply:
see the above comment to see why I decided to tag the tokens like this.

Reply:
refactored in [18ed767](https://github.com/JanssenProject/jans/pull/10436/commits/18ed7673bc485b13e045d2159b04de12cfc8641a)